File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/dto/AccessResult.java
package ru.tggc.telegrambotcore.dto;

import jakarta.annotation.Nullable;

public record AccessResult(boolean allowed, @Nullable Response response) {

    public static AccessResult allow() {
        return new AccessResult(true, null);
    }

    public static AccessResult deny(Response response) {
        return new AccessResult(false, response);
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/dto/AccessResult.kt
package ru.tggc.telegrambotcore.dto

@JvmRecord
data class AccessResult(val allowed: Boolean, val response: Response?) {
    companion object {
        @JvmStatic
        fun allow(): AccessResult {
            return AccessResult(true, null)
        }

        @JvmStatic
        fun deny(response: Response?): AccessResult {
            return AccessResult(false, response)
        }
    }
}
